bin/sync_vsc_email_postfix.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+#
+# Copyright 2013-2019 Ghent University
+#
+# This file is part of vsc-administration,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-administration
+#
+# All rights reserved.
+#
+"""
+This script creates the list of canonical email adresses for VSC users.
+"""
+from __future__ import print_function
+
+import logging
+
+from vsc.accountpage.sync import Sync
+
+
+class VscPostfixSync(Sync):
+    CLI_OPTIONS = {
+        'postfix_alias_map': ('Location of the postfix canonical map', None, 'store', '/etc/postfix/vsc_canonical'),
+    }
+
+    def do(self, dryrun):
+        """Actual work.
+
+        - retrieve all the email addresss from changed users
+        - read the current file
+        - remove inactive users' email addresses
+        - add new/changed email addresses
+        - write file
+        """
+        active_accounts, inactive_accounts = self.get_accounts()
+
+        active_emails = dict([(a.vsc_id, a.email) for a in active_accounts])
```

Comment:
why no set for both active and inactive mails, why not work with sets only?

Reply:
How does one update a set of tuples where the key is the first tuple element in Python?

Reply:
ah, yes. i was confused with the copy paste below

but then the `-=` for inactive users also does not work (imagine a user that modified email and became inactive).
so the code should use same objects for everything.

for inactive users, you only need a set of the vsc accounts, not their email.
address_set should abe a dict too, and you can already filter out the inatcive users in the readlines loop

Reply:
Ah yes, I see your point. It's a small window, but possible indeed.